resources/js/models/Story.ts

```diff
@@ -0,0 +1,67 @@
+import User from './User';
+import axios from 'axios';
+import fakeStory from './story.json';
+import { config } from '../config';
+import {isToday, isYesterday, format} from 'date-fns';
+
+const mainUrl = `${config.host}/api/stories`;
+export default class Story {
+    id: number;
+    image: {
+        id: number;
+        url: string;
+    };
+    createdAt: string;
+    user: User;
+
+    constructor(json: {
+        id: number;
+        image: {
+            id: number;
+            url: string;
+        };
+        createdAt: string;
+        user: User;
+    }) {
+        this.id = json.id;
+        this.image = json.image;
+        this.createdAt = json.createdAt;
+        this.user = json.user;
+    }
+
+    static fromJson(json: any): Story {
+        return new Story(json);
+    }
+
+    static async all(): Promise<Story[]> {
+            try{
+                return fakeStory.map((story) => Story.fromJson(story));
+            }
+            catch{
+                throw new Error('Error fetching users')
+            }
+    }
+
+      getFormattedDate = () => {
+        if (isToday(this.createdAt)) {
+            return `Toady at ${format((this.createdAt), 'HH:mm')}`
+        }
+        if (isYesterday(this.createdAt)) {
+            return `Yesterday at ${format((this.createdAt), 'HH:mm')}`
+        }
+        return format((this.createdAt), 'dd/MM/yyyy')
+    };
+
+    getFormattedStory = () => {
```

Comment:
no no no no! 
there is no place for 'component specific code' inside your model

it is okey to write non-business logic in your model as long as its 'decoupled' with your components

remember, business logic is the logic that sits between your backend and your components and is responsible with working and preparing with the data that backend sends you.

here what you did is you have coupled your model with your components, and what coupling means is that when you change this code inside this model, it will brake your components, so you made the model and the component tightly dependent on each other or another words you made the 'couple'!

this logic must completely remove from here and find another way to implement this, you should not be needing such logic anyway this simply means you have gone wrong way other places so you needed this!

Reply:
The main project was written like that, and the date-fns library was used, so I studied how to use it to show the time the story was published.
Should I totally rewrite it or just write it somewhere else??